test_scripts/API/VehicleData/GearStatus/common.lua

```diff
@@ -0,0 +1,374 @@
+--! @parameters:
+--! pHashValue: application's hashId
+--! pAppId: application number (1, 2, etc.)
+--! @return: none
+--]]
+function m.setHashId(pHashValue, pAppId)
+  hashId[pAppId] = pHashValue
+end
+
+--[[ @getHashId: Get hashId of an app which is required during resumption
+--! @parameters:
+--! pAppId: application number (1, 2, etc.)
+--! @return: app's hashId
+--]]
+function m.getHashId(pAppId)
+  return hashId[pAppId]
+end
+
+--[[ @checkParam: Check the absence of unexpected params in GetVehicleData and OnVehicleData on the mobile app side
+--! @parameters:
+--! pData - parameters for mobile response/notification
+--! pRPC - RPC for mobile request/notification
+--! @return: true - in case response/notification does not contain unexpected params, otherwise - false with error message
+--]]
+function m.checkParam(pData, pRPC)
+  local count = 0
+  for _ in pairs(pData.payload.gearStatus) do
+    count = count + 1
+  end
+  if count ~= 1 then
+    return false, "Unexpected params are received in " .. pRPC
+  else
+    return true
+  end
+end
+
+--[[ @gearStatus: Clone table with data for use to GetVD and OnVD RPCs
+--! @parameters: none
+--! @return: table for GetVD and OnVD
+--]]
+function m.getGearStatusParams()
+  return utils.cloneTable(gearStatusData)
+end
+
+--[[ @responseTosubUnsubReq: Clone table with data for use in SubscribeVD and UnsubscribeVD RPCs
+--! @parameters: none
+--! @return: table for SubscribeVD and UnsubscribeV
+--]]
+function m.responseTosubUnsubReq()
+  return utils.cloneTable(m.subUnsubResponse)
+end
+
+--[[ @setValue: Set value for params from `gearStatus` structure
+--! @parameters:
+--! pParam: parameters from `gearStatus` structure
+--! pValue: value for parameters from the `gearStatus` structure
+--! @return: table for GetVD and OnVD
+--]]
+function m.setValue(pParam, pValue)
+  local param = m.gearStatus()
+  param[pParam] = pValue
+  return param
+end
+
+--[[ @getVehicleData: Processing GetVehicleData RPC
+--! @parameters:
+--! pData: parameters for GetVehicleData response
+--! pParam: parameter for GetVehicleData request
+--! @return: none
+--]]
+function m.getVehicleData(pData, pParam)
+  if not pParam then pParam = "gearStatus" end
+  local cid = m.getMobileSession():SendRPC("GetVehicleData", { [pParam] = true })
+  m.getHMIConnection():ExpectRequest("VehicleInfo.GetVehicleData", { [pParam] = true })
+  :Do(function(_,data)
+    m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { [pParam] = pData })
+  end)
+  m.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS", [pParam] = pData })
+end
+
+--[[ @subUnScribeVD: Processing Subscribe/UnsubscribeVehicleData RPC
+--! @parameters:
+--! pRPC: RPC for mobile request
+--! pParam: parameters for Subscribe/UnsubscribeVehicleData RPC
+--! pVDType: VehicleDataType value
+--! isRequestOnHMIExpected: true or omitted - in case VehicleInfo.Subscribe/UnsubscribeVehicleData_request on HMI is expected, otherwise - false
```

Comment:
```suggestion
--! isRequestOnHMIExpected: true or omitted - in case VehicleInfo.Subscribe/UnsubscribeVehicleData_request on HMI is expected, otherwise - not expected
```